dart/dynamics/Entity.h

```diff
@@ -63,11 +63,23 @@ class Shape;
 /// may have different policies about how/if their reference frame or name
 /// can be changed. Use the Detachable class to create an Entity whose reference
 /// Frame can be changed arbitrarily.
-class Entity
+class Entity : public common::Subscription
 {
 public:
   friend class Frame;
 
+  enum Notification {
```

Comment:
How about to use `enum class` as
```cpp
enum class Notification : int {
  // items
}
```
for [type safety and scoped enumeration](http://www.cprogramming.com/c++11/c++11-nullptr-strongly-typed-enum-class.html)?